### PR TITLE
Terrain estimate: Added height above ground source bitmask

### PR DESCRIPTION
**Pull request by bresch**
A PR in Firmware will follow this to actually log the bitmask

### CODE REVIEW

File: EKF/terrain_estimator.cpp [hunk 1]
```diff
@@ -289,6 +290,9 @@ void Ekf::updateTerrainValidity()
 						    && !_control_status.flags.opt_flow;
 
 	_hagl_valid = (_terrain_initialised && (recent_range_fusion || recent_flow_for_terrain_fusion));
+
+	_hagl_sensor_status.flags.range_finder = recent_range_fusion && (_time_last_fake_hagl_fuse != _time_last_hagl_fuse);
```
Comment (kamilritz) [inline EKF/terrain_estimator.cpp:294]: Why is the second condition necessary?

Comment (RomanBapst) [reply EKF/terrain_estimator.cpp:294]: @kamilritz The idea of reporting this information is that the system can determine whether an actual range finder is in use or not. If the system is on the ground then the terrain estimate is valid but it does not tell us yet if a range finder is available. Distinguishing between fake range data (which really does not stem from a sensor but just from the assumption being on the ground) allows the system to understand when a range finder is in use.

Comment (RomanBapst) [reply EKF/terrain_estimator.cpp:294]: @kamilritz I would be ok to actually add a "fake_range_data" or "on_ground_assumption" flag if you think it's useful.

**Comment by bresch** [comment]
I'll merge it as is and we can still patch it if needed.

File: EKF/terrain_estimator.cpp (hunk 1 above)
Comment (kamilritz) [reply EKF/terrain_estimator.cpp:294]: @RomanBapst That would be nice. The "on_ground_assumption" sounds very clean

**Pull request merged**